src/app/components/pmapi/pmapi.service.js

```diff
@@ -161,11 +160,34 @@
                 });
         }
 
-        function convertTimestampToMillis(response) {
-            // timestamp is in milliseconds
+        function convertTimestampToMillisOrMicros(response) {
             var timestamp = (response.data.timestamp.s * 1000) +
                 (response.data.timestamp.us / 1000);
             var values = response.data.values;
+			
+			if ($rootScope.metadata == undefined) {
+				return {
+                timestamp: timestamp,
+                values: values
+            	};
+			}
+			
+			var units = [];
+			response.data.values.forEach(function(ele) {
+				units.push($rootScope.metadata.find(function(element) {
+					return element.name == ele.name;
+				}).units);
+			})
+			
+			units.forEach(function(ele) {
```

Comment:
This is not efficient, since we would have to do a `unit` lookup for every `metric`, every time there is an update. A more efficient solution would likely store the unit along with the `metric` so the lookup doesn't have to be performed.